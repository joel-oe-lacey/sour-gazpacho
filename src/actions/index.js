export const loadMovies = (movies) => ({ 
    type: 'LOAD_MOVIES',
    movies 
});

export const loginUser = (user) => ({
    type: 'LOGIN_USER',
    user
})

export const logOut = () => ({
    type: 'LOG_OUT',
    user: {}
})