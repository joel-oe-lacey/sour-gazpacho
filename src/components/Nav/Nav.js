import React from 'react';
import './Nav.scss';

const Nav = () => {
    return (
        <nav className="nav">
            
        </nav>
    )
}

export default Nav;