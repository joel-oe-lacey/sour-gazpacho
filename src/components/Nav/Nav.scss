@import '../../variables';

.nav {
  height: 60px;
  width: 100%;
  background-color: $primary-yellow-color;
  display: grid;
  grid-template-columns: repeat(3, 33%);
  padding: 1%;
}

.nav-link {
  height: 100%;
  width: 20%;
  text-decoration: none;
  text-align: center;
  background-color: $primary-red-color;
  color: $primary-yellow-color;
  font-size: 1.2em;
  font-weight: bold;
  margin-left: 5%;
  border: 1px solid $primary-red-color;
  border-bottom-right-radius: 5% 40%;
  border-top-right-radius: 5% 40%;
  border-bottom-left-radius: 5% 40%;
  border-top-left-radius: 5% 40%;
  transition: box-shadow 300ms;
  box-shadow: none;

  &:hover {
      cursor: pointer;
      box-shadow: 0.1rem 0.3rem 0.6rem $primary-gray-color;
  }
}

#tomato-splatter {
  left: 80%;
  height: 180px;
  margin-top: -40px;
  position: absolute;
}

h1 {
  font-size: 1.5em;
  color: $primary-red-color;
}
