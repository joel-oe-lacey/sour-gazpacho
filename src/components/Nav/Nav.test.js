import { Nav, mapStateToProps, mapDispatchToProps } from './Nav.js';
import { logOut } from '../../actions';
import { shallow } from 'enzyme';
import React from 'react';



describe('Nav', () => {
  let mockUser = {name: 'Karl the Llama', id: 9, password: 'password1'}

  describe('Nav Component', () => {
    let wrapper;
    let mockLogOutFromStore = jest.fn()

    beforeEach(() => {
    wrapper=shallow(<Nav
        user={mockUser}
        logOutFromStore={mockLogOutFromStore}
      />);
    })

    it('should be an instance of Nav', () => {
      expect(wrapper).toMatchSnapshot();
    })

  })
  describe('mapStateToProps', () => {
    it('should return an object with an array of todos', () => {
      const mockState = {
        user: mockUser,
        movies: 'Big Lebowski'
      }
      const expected = { user: mockUser }
      const mappedProps = mapStateToProps(mockState)
      expect(mappedProps).toEqual(expected)
    })

  })
  describe('mapDispatchToProps', () => {
    it('should call dispatch with addTodo action when logOutFromStore is called', () => {
      const mockDispatch = jest.fn();
      const actionToDispatch = logOut();
      const dispatchedProps = mapDispatchToProps( mockDispatch );
      dispatchedProps.logOutFromStore();
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    })
  })
})
