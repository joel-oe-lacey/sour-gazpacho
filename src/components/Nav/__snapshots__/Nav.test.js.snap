// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav Nav Component should be an instance of Nav 1`] = `
<nav
  className="nav"
>
  <Link
    to="/"
  >
    <button
      onClick={[MockFunction]}
    >
      Log Out
    </button>
  </Link>
  <h1>
    Sour Gazpacho
  </h1>
  <img
    alt="kill bill style blood splatter"
    id="blood-splatter"
    src="blood-splatter.png"
  />
</nav>
`;
